tests/test_interfaces.py

```diff
@@ -778,6 +723,103 @@ def test_get_interface_mac_address(self):
 
         self.assertEqual(self.network_manager.get_interface_mac(interface_name), self.mac_address)
 
+    def test_get_interface_1_ap_1_mon_internet_interfaces(self):
```

Comment:
The thing about unit testing is that we should only test a single unit in a test. In this test case the title indicates that you are testing the `get_interface` method while `get_interface` is never called. All I see is 3 calls to the `is_interface_valid`. Can you explain what is that you are testing in here?